pymdownx/blocks/block.py

```diff
@@ -130,13 +135,13 @@ def type_html_identifier(value):
     return m.group(0)
 
 
-def _delimiter(string, split, string_type):
+def _delimiter(string: Any, split: str, string_type: Callable[[Any], str]) -> list[str]:
     """Split the string by the delimiter and then parse with the parser."""
 
     l = []
     # Ensure input is a string
-    string = type_string(string)
-    for s in string.split(split):
+    _string = type_string(string)
```

Comment:
my lsp was happier with a new name for the type-restricted value, I'm not sure that it's strictly required by mypy. I think it makes things more explicit, but I'm happy to revert.

Reply:
It's not because of the rename, it's because the input is typed as any and then the function returns string to the same variable. Using a new variable ensures causes the new variable to be scooped to string only.